tests/core/test_graph.py

```diff
@@ -649,12 +649,14 @@ def test_benchmark_get_features(benchmark, num_types, type_arg, feature_size):
         # leave the argument as None, and so use inference of the type
         node_type = lambda ty: None
 
+    ty, all_ids = random.choice(ty_ids)
```

Comment:
ah good point about the between-run variance, I think your sugestion is good for now. In the furture I reckon something like this could work:

```python
@pytest.mark.parametrize("use_ilocs",  [False, True])
def test_benchmark_get_features(..., use_ilocs):
    ...
    ty, all_ids = random.choice(ty_ids)
    all_selected_ids = [random.choices(all_ids, k=SAMPLE_SIZE) for _ in range(number_of_runs)]
    all_selected_ilocs = [sg.node_ids_to_ilocs(selected_ids) for selected_ids in all_selected_ids]
    counter = 0
    def f():
        
        if use_ilocs is True:
            selected_ilocs = all_selected_ilocs[counter]
            sg.node_features(selected_ilocs, node_type(ty), use_ilocs=True)
        elif use_ilocs is False:
            selected_ids = all_selected_ids[counter]
            sg.node_features(selected_ids, node_type(ty), use_ilocs=False)

        counter += 1        
```

But it seems a bit fiddly and un-pythonic, so I'll go with yours 👍